### DISCUSSION TITLE
Artifacts on Volume plot - but not on slice, same data

### DISCUSSION THREAD

**Discussion by KestutisMa** [marcomusy/vedo#1029] [Q&A] [answered]
Dear Vedo Team,
I love how intuitive and flexible Vedo is.
But I have observed some artifacts on volume plot, but not on slice plot of same data.

1) Also, I am using Vedo v2023.5.0, also tired to copy volume_slice function from master branch. And to use slice_plane(... mode='nearest') but view looks like interpolator is 'cubic' or 'linear' not 'nearest'.

2) Artifacts:

Volume+Slice:
![image](https://github.com/marcomusy/vedo/assets/29224758/d1d64858-9641-4609-acdd-c9c530919570)

Slice:
![image](https://github.com/marcomusy/vedo/assets/29224758/83cc4c27-01a8-44e0-8383-bed267989d9f)

I've done good bit of digging code and docs without any luck.
I would very appreciate it if someone could help me!

**Comment by marcomusy** [comment]
Hi Thanks for reporting, can you explain what is the artifact you refer to? Is it the jagged line:
![Screenshot from 2024-01-19 16-26-23](https://github.com/marcomusy/vedo/assets/32848391/b61a5938-82b3-451a-b72f-2a70f54aa710)

if so I think this is due to antialiasing.. for reasons I don't even understand ;) the `force_translucent()` seems to fix that, but the plane also looks dimmer (which might be infact not wrong..):

```py
from vedo import *
X, Y, Z = np.mgrid[:30, :30, :30]
scalar_field = ((X-15)**2 + (Y-15)**2 + (Z-15)**2) /225
vol = Volume(scalar_field)
vslice = vol.slice_plane((5,0,0), normal=(1,0,0))
vslice.force_translucent()
show(vslice, vol, viewup='z').close()
```

![Screenshot from 2024-01-19 16-42-19](https://github.com/marcomusy/vedo/assets/32848391/f4fac824-1d54-4189-a539-923aa508f7c3) ![Screenshot from 2024-01-19 16-42-34](https://github.com/marcomusy/vedo/assets/32848391/f35e3069-b9d5-463d-bfab-f29cd069052d)

**Reply by KestutisMa** [reply]
It's some ghostly contours (discontinuities), that appear on volume, but not on slice.
![image](https://github.com/marcomusy/vedo/assets/29224758/74e1d764-7c9f-4b80-9d3e-db51bd55c965)

Thanks on note about AA

**Reply by marcomusy** [reply]
For that you can try:
```
vol = Volume(...).jittering(True)
```

and/or you can change ["mode"](https://vedo.embl.es/autodocs/content/vedo/vedo/visual.html#VolumeVisual.mode)
```
vol = Volume(...).mode(1)
```

**Reply by KestutisMa** [reply]
thank you, .mode(1) looks right:
![image](https://github.com/marcomusy/vedo/assets/29224758/0b990d84-f80d-4433-97d6-5a5dd5435a61)

Wondering why mode(0) is giving such unexpected results? also, jittering(true) doesn't change results in mode(0)

**Reply by marcomusy** [reply]
Are you sure it's an artifact? By looking carefully at your snapshot that could be a real change in the scalar (you may check it by looking at the isosurface):
https://github.com/marcomusy/vedo/blob/master/examples/volumetric/app_isobrowser.py

**Reply by KestutisMa** [reply]
isobrowser is very convenient, I embedded it to my test code. Tried to slide all isovalues and indeed observed same discontinuities in volume as in previous post.
Wondering why slice_plane shows just very slight changes in color (in data array there are 30 values along vertical axis), where Volume with .mode(0), gives steep discontinuities in color at that places?
Maybe because of my color blindness. :smile:

**Comment by marcomusy** [comment, marked as answer]
> Maybe because of my color blindness. 😄

LOL that might explain it... but indeed the difference is quite hard to see in the viridis colormap, maybe "[gist_ncar](https://github.com/marcomusy/vedo/blob/master/docs/colormaps.jpg)" or some other rapidly changing hue cmap can help identifying it.

**Reply by KestutisMa** [reply]
It's [cividis](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0199239) specially designed colormap _"for the most accurate perception of scientific data by as many viewers as possible."_, I was hoping it look more clear.
